### DISCUSSION TITLE
jj git clone --origin NAME

### DISCUSSION THREAD

**Discussion by samueltardieu** [jj-vcs/jj#4447] [General]
Would it be useful to add `--origin NAME` (or `--remote NAME`) to use another name than `origin` as the default remote?

My use case is that I often participate to projects hosted on GitHub. I call the upstream remote `upstream`, and when I contribute my own fork is `origin`. I track the `main` branch` from `upstream`, and private branches from origin.

I would like to be able to do:

```bash
$ jj git clone --origin upstream https://github.com/…
$ jj git remote add origin git@github.com:samueltardieu/…
```

Right now, I have to create and clone my fork (even though I'm not sure I'll contribute yet), and then add the `upstream` remote, untrack `main` and track `main@upstream`. Or I must clone the upstream, rename `origin` as `upstream`, forget about `main` and track `main@upstream`, then I can add my `origin` remote when it is needed.

I have the feeling that being able to set the original remote name (as in `git clone --origin NAME https://…`) would allow smoother processes.

`--origin NAME` could also be `--remote NAME`.  `--origin` is how git calls it, `--remote` is the standard arg name used by jj.

If people think this might be useful, I might devise a PR containing this feature.

**Comment by emilazy** [comment]
FWIW I used to do the `origin`/`upstream` thing, but have found I am much happier making `origin` always the upstream and adding an `emily` remote for my own forks. I set `git.fetch` to `["origin", "emily"]`, `git.push` to `"emily"`, always use HTTPS URLs, and put this in my Git config:

```
[url "git@github.com:"]
    pushInsteadOf = "https://github.com/"
```

I know this isn’t really a direct solution to your problem, but it’s great. No regrets. Try it out.

**Reply by samueltardieu** [reply]
Hi Emily.

Happy to see you're satisfied with those names. I've tried this naming scheme in the past, but it didn't really catch with me. I set my `git.{fetch,push}` in a similar way as yours (I even think I'm the one who implemented this in jj, ages ago).

Incidentally, I didn't know of the url substitution in the git config, I'll have a closer look, thanks!

**Comment by samueltardieu** [comment]
I have opened https://github.com/martinvonz/jj/pull/4449 to test the idea.
